### ISSUE TITLE
Some questions on training MatterGen from scratch

### ISSUE THREAD

**Issue by yuedawang** [microsoft/mattergen#172]
Dear developer,

When I learn how to train MatterGen from scratch according to the section of 'Train MatterGen yourself' in README.md, I met some questions:

1. The dataset for this training is mp_20 downloading from here (https://github.com/microsoft/mattergen/tree/main/data-release/mp-20). My training command is: `mattergen-train data_module=mp_20 ~trainer.logger trainer.accumulate_grad_batches=20`. The training process might last for 899 epochs according to the end output of the job log as:
```
Epoch 899:  97%|█████████▋| 1050/1086 [02:10<00:04,  8.07it/s, v_num=0, loss_train_step=0.190, pos_train_step=1.660, cell_train_step=0.0238, loss_train_epoch=0.229, pos_train_epoch=2.120, cell_train_epoch=0.0172]
Epoch 899: 100%|██████████| 1086/1086 [02:15<00:00,  8.04it/s, v_num=0, loss_train_step=0.190, pos_train_step=1.660, cell_train_step=0.0238, loss_train_epoch=0.229, pos_train_epoch=2.120, cell_train_epoch=0.0172]
Epoch 899: 100%|██████████| 1086/1086 [02:15<00:00,  8.04it/s, v_num=0, loss_train_step=0.259, pos_train_step=2.470, cell_train_step=0.0121, loss_train_epoch=0.229, pos_train_epoch=2.120, cell_train_epoch=0.0172]

Validation: 0it [00:00, ?it/s]

Validation:   0%|          | 0/142 [00:00<?, ?it/s]

Validation DataLoader 0:   0%|          | 0/142 [00:00<?, ?it/s]

Validation DataLoader 0:  35%|███▌      | 50/142 [00:06<00:12,  7.17it/s]

Validation DataLoader 0:  70%|███████   | 100/142 [00:13<00:05,  7.15it/s]

Validation DataLoader 0: 100%|██████████| 142/142 [00:19<00:00,  7.13it/s]
Epoch 899: 100%|██████████| 1086/1086 [02:35<00:00,  6.99it/s, v_num=0, loss_train_step=0.259, pos_train_step=2.470, cell_train_step=0.0121, loss_train_epoch=0.229, pos_train_epoch=2.120, cell_train_epoch=0.0172]

                                                                          
Epoch 899: 100%|██████████| 1086/1086 [02:42<00:00,  6.68it/s, v_num=0, loss_train_step=0.259, pos_train_step=2.470, cell_train_step=0.0121, loss_train_epoch=0.232, pos_train_epoch=2.150, cell_train_epoch=0.0172]`Trainer.fit` stopped: `max_epochs=900` reached.

Epoch 899: 100%|██████████| 1086/1086 [02:42<00:00,  6.67it/s, v_num=0, loss_train_step=0.259, pos_train_step=2.470, cell_train_step=0.0121, loss_train_epoch=0.232, pos_train_epoch=2.150, cell_train_epoch=0.0172]
```
It seems like the training already reached the maximum epochs as '`Trainer.fit` stopped: `max_epochs=900` reached.' in the log file. While according to the README.md, 'The validation loss (loss_val) should reach 0.4 after 360 epochs (about 80k steps). ', so should I stop the training job when the loss (loss_val) once reach or less than 0.4? How to find the real-time value of loss_val? 

2. For larger dataset like alex_mp_20, could this criteria (the loss (loss_val) reach or less than 0.4) also be valid for stopping training? In addition, could the small loss_val (<0.4) be the only criteria for a qualified training?

3. After finishing training, there are two folders in the `outputs/singlerun/2025-06-07` as `14-19-36` and `14-22-35`, I guess the later one of `14-22-35` might contains the lastest training results. And I found the checkpoint files in two places, one is in `outputs/singlerun/2025-06-07/14-22-35/checkpoints` as `'epoch=899-step=49500.ckpt'`; another is in `outputs/singlerun/2025-06-07/14-22-35/lightning_logs/version_0/checkpoints` as `'epoch=884-loss_val=0.25.ckpt'` and `last.ckpt`. So, which checkpoint files should I choose as the final model for sampling? In addition, according to the filename of `'epoch=884-loss_val=0.25.ckpt'`, the loss_val might already reached to 0.25 at 884 epoch, maybe I should stop the training earlier? (It comes back the first question of how to find the real-time loss_val during the training process.) 

4. For crystal structure prediction (CSP) tasks, what kind of features of the dataset determine the final model quality? More atoms in one structure or more total structures? For example, there are two datasets of mp_40 (contains less than 40 atoms in one structure, https://github.com/hspark1212/chemeleon/tree/main/data/mp-40) and alex_mp_20 (more total structures but each structures contains less than 20 atoms, https://github.com/microsoft/mattergen/tree/main/data-release/alex-mp), which dataset could be trained to obtain better model for CSP task?

Looking forward for your kind reply!
